Assets/Scripts/CommonNPC/EntityScanner.cs

```diff
@@ -45,20 +45,20 @@ public int ObstacleMask
         set => _obstacleMask = value;
     }
 
+    private Collider[] _candidates = new Collider[16]; // Adjust size as needed, e.x. 16 friendly NPCs in one space
+
+
     public IEnumerable<Collider> doScan(int maxObjects = 0)
     {
         var eyePosition = SourceTransform.position + this.EyeOffset;
 
-        //Vector3 boxCenter = SourceTransform.position + (SourceTransform.forward * (ViewDistance / 2f));
-        //Vector3 boxHalfExtents = new Vector3(ViewDistance / 2f, 20.5f, ViewDistance / 2f);
-        // Collider[] candidates = Physics.OverlapBox(boxCenter, boxHalfExtents, SourceTransform.rotation, _targetMask);
+        _candidates = null; // Reset candidates array
```

Comment:
I'm confused, isn't this destroying the object? Won't this generate an error?

Reply:
I guess you don't really need to clear it since it's fixed size, instead you probably want something like:

```
for (int idx = 0; idx < candidateCount; idx++)
{
  // do stuff
}
```

Reply:
It would probably be quicker to just use `candidateCount` in the for-loop -- then you don't have to call anything. But clear is fine too.

Reply:
Actually, it may be a good idea to use `candidateCount` now that I think about it. Every (idle) NPC, and probably many others, are going to be using `doScan()` quite a lot.

I'll merge this and fool around with it tho.